packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return Promise.all([
+    knex.schema.renameTable('paymentPointers', 'walletAddresses'),
+    knex.schema.renameTable('paymentPointerKeys', 'walletAddressKeys'),
+    knex.schema.alterTable('walletAddressKeys', function (table) {
+      table.renameColumn('paymentPointerId', 'walletAddressId')
+      table.foreign('walletAddressId').references('walletAddresses.id')
+    }),
+    knex.schema.alterTable('quotes', function (table) {
+      table.renameColumn('paymentPointerId', 'walletAddressId')
+      table.foreign('walletAddressId').references('walletAddresses.id')
```

Comment:
I'm not actually sure, what happens to the existing `paymentPointer.id` foreign key here? Does it get properly handled in the rename, or dropped?

Reply:
This also applies the indexes (not pictured above) in  line 16: `table.index(['walletAddressId', 'createdAt', 'id'])`.

Looks like the migration is duplicating these. Two fkey constraints with different fkey names but the same relationship. I think we can just remove the `table.foreign('walletAddressId').references('walletAddresses.id')` line (and the indexes). This is what I see in `Indexes` and `Foreign-key constraints:` from doing a `\d quotes` after this migration. 

quote:

```
Indexes:
    "quotes_pkey" PRIMARY KEY, btree (id)
    "quotes_paymentpointerid_createdat_id_index" btree ("walletAddressId", "createdAt", id)
    "quotes_walletaddressid_createdat_id_index" btree ("walletAddressId", "createdAt", id)
Foreign-key constraints:
    "quotes_assetid_foreign" FOREIGN KEY ("assetId") REFERENCES assets(id)
    "quotes_feeid_foreign" FOREIGN KEY ("feeId") REFERENCES fees(id)
    "quotes_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    "quotes_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
 ```
 
 and here is some of \d "walletAddresses" for good measure (see the duplicate quote fkey):
```
Referenced by:
    TABLE ""incomingPayments"" CONSTRAINT "incomingpayments_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE ""incomingPayments"" CONSTRAINT "incomingpayments_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE ""outgoingPayments"" CONSTRAINT "outgoingpayments_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE ""outgoingPayments"" CONSTRAINT "outgoingpayments_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE ""walletAddressKeys"" CONSTRAINT "paymentpointerkeys_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE "quotes" CONSTRAINT "quotes_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE "quotes" CONSTRAINT "quotes_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
    TABLE ""walletAddressKeys"" CONSTRAINT "walletaddresskeys_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)
```

Reply:
Actually instead of removing these lines we might want to just drop the old one. This way we end up with `"quotes_walletaddressid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id)` instead of "quotes_paymentpointerid_foreign" FOREIGN KEY ("walletAddressId") REFERENCES "walletAddresses"(id).

So doing like:

```
    knex.schema.alterTable('quotes', function (table) {
      table.dropForeign(['paymentPointerId'])
      table.dropIndex(['paymentPointerId', 'createdAt', 'id'])
      table.renameColumn('paymentPointerId', 'walletAddressId')
      table.foreign('walletAddressId').references('walletAddresses.id')
      table.index(['walletAddressId', 'createdAt', 'id'])
    }),
 ```